ref/Dictionaries.md

```diff
@@ -1,24 +1,213 @@
+
+###  Elementary Dictionary Operations
+
+**Description:**
+Without arguments, this creates an empty dictionary.
+
+    > {}
+    < {}
+
+------
+
+### Creating a simple dictionary: `{"key" : val, ...}`
+
+**Description:**
+The expression `{"key1" : val, ...}` created a basic dictionary. 
+
+    > {"a" : 1, "b" : 2};
+    < {a:1, b:2}
+
+------
+
+### Printing a dictionary
+
+**Description:**
+As usual the function `print` will output the object. This will generally be not a valid JSON object: therefore one can use the modifier `JSON`. The output then can be used by a JSON parser.
+
+    > print({"a" : 1, "b" : false, "c": "mystring"}, JSON->true);
+    * {"a":1,"b":false,"c":"mystring"}
+
+**Circular Dictionaries**
+Circular objects are allowed, but a printout will be shortened. The default recursion depth is 1000 but can be adjusted with the modifier `maxDepth`:
+
+    > dic={"a" : 1}; dic.b = dic; print(dic, maxDepth->3);
+    * Warning: We visited a key-value pair very often or encountered a very deeply nested dictionary. Dictionary is probably cyclic. Output will be probably incomplete.
+    * {a:1, b:{a:1, b:{a:1, b:{a:..., b:...}}}}
+
+
+------
+
+### Accessing Elements of Lists: `‹dict›.name`
+
+**Description:**
+Basic (static) access is possible via the dot operator. The variable after the dot is **not** evaluated.
+
+    > json = {"a" : 1, "b" : 2};
+    > json.a
+    < 1
+
+------
+### Retrieving all keys and values of a dictionary: `keys(‹dict›)` and `values(‹dict›)`
+
+**Description:**
+To retrieve all keys or values of a dictionary as a list one can use `keys(‹dict›)` and `values(‹dict›)`
+
+
+    > json = {"a" : 1, "b" : 2};
+    > keys(json);
+    < ["a", "b"]
+    > values(json);
+    < [1, 2]
+
+------
+
+### Dynamically accessing Elements of Lists: `‹dict›_‹var›` and `take(‹dict›,‹var›)`
+
+**Description:**
+One can dynamically access the individual elements of a dict either with the infix operator `‹list›_‹var›` or the functional operator `take(‹list›,‹var›)`. The keys (`var`) can be of any printable type.
+
+
+    > json = {"a" : 1, "b" : 2};
+    > json_"a"
+    < 1
+    > take({"a" : 1, "b" : 2}, "b")
+    < 2
+
+------
+
+### Applying an expression: `apply(‹dict›,‹expr›)`
+
+**Description:**
+This operator generates a new list by applying the operation `‹expr›` to all elements of a dict (but not to the keys) and collecting the results.
+As usual, `#` is the run variable, which successively takes the value of each element in the list.
+
+    > apply({"a" : 1, "b" : 2}, #^2);
+    < {a:1, b:4}
+
+------
+
+### Applying an expression: `apply(‹dict›,‹var›,‹expr›)`
+
+**Description:**
+Similar to `apply(‹list›,‹expr›)`, but the run variable is now named `‹var›`.
+The variable is local to the expression.
+
+    > v = 123;
+    > apply({"a" : 1, "b" : 2}, v, v^2);
+    < {a:1, b:4}
+    > v
+    < 123
+
+------
+
+
+### Applying an expression: `apply(‹dict›,‹varVal›, ‹varKey›,‹expr›)`
```

Comment:
could we have apply$3 with ` iterator->"pair"` such as it is done with forall for this purpose?
(anyway, the order first ‹varVal› and then ‹varKey› and then misleading)